File: containerwithmostwater.py
class Solution:
    def maxArea(self, height:list[int])->int:
        l, r =0, len(height)-1
        res = 0
        while l<r:
            res = max(res, min(height[l], height[r])*(r-l))
            if height[l]<height[r]:
                l+=1
            elif height[l]>=height[r]:
                r-=1

        return res

if __name__=="__main__":
    d =  Solution()
    du = d.maxArea([2,7,9,3,6,4,9])
    print("Max Area :{}".format(du))

File: Java/IntelliJ/Demo/src/main/java/org/example/ContainerWithMostWater.java
package org.example;

public class ContainerWithMostWater {
    public static int maxArea(int[] heights){
        int maxArea =0;
        int left=0,right=heights.length-1;
        while(left<right){
            int curArea = (right - left)*Math.min(heights[right], heights[left]);
            maxArea = Math.max(curArea, maxArea);
            if(heights[right]<heights[left]){
                right-=1;
            }else{
                left+=1;
            }
        }
        return maxArea;
    }

    public static void main(String[] args){
        // two pointer
        // left, right, maxArea, current Area
        // left>right right-- else left-- else left--
        int maxArea = maxArea(new int[]{1,8,6,2,5,4,8,3,7});
        System.out.println("Container with Most water: "+maxArea);
    }
}
